samples/c/CMakeLists.txt

```diff
@@ -26,6 +26,11 @@ if(BUILD_EXAMPLES AND OCV_DEPENDENCIES_FOUND)
     add_executable(${the_target} ${srcs})
     target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} ${OPENCV_C_SAMPLES_REQUIRED_DEPS})
 
+    if(UNIX)
+        find_package (Threads)
+        target_link_libraries(${the_target} "${CMAKE_THREAD_LIBS_INIT}" m)
+    endif()
```

Comment:
it's better to extract this code to `samples/CMakeLists.txt` and set libs to `OPENCV_LINKER_LIBS` if it's not defined yet:
```
if (NOT DEFINED OPENCV_LINKER_LIBS)
    if(UNIX)
        find_package (Threads)
        set(OPENCV_LINKER_LIBS "${CMAKE_THREAD_LIBS_INIT}" m)
    endif()
endif()
```